docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -241,236 +287,445 @@ This generates the ABI code and moves it to `./contract-examples/contracts` as `
-9- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
-10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+1- Set a suitable config key in generated module.go. E.g: "yourPrecompileConfig"
+2- Read the comment and set a suitable contract address in generated module.go. E.g:
+	ContractAddress = common.HexToAddress("ASUITABLEHEXADDRESS")
+3- It is recommended to only modify code in the highlighted areas marked with "CUSTOM CODE STARTS HERE". Typically, custom codes are required in only those areas.
+Modifying code outside of these areas should be done with caution and with a deep understanding of how these changes may impact the EVM.
+4- Set gas costs in generated contract.go
+5- Force import your precompile package in precompile/registry/registry.go
+6- Add your config unit tests under generated package config_test.go
+7- Add your contract unit tests undertgenerated package contract_test.go
+8- Additionally you can add a full-fledged VM test for your precompile under plugin/vm/vm_test.go. See existing precompile tests for examples.
+9- Add your solidity interface and test contract to contract-examples/contracts
+10- Write solidity tests for your precompile in contract-examples/test
+11- Create your genesis with your precompile enabled in tests/e2e/genesis/
+12- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+13- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh
 */
 ```
 
 <!-- markdownlint-enable MD013 -->
 
-### Step 1: Set Contract Address
+### Step 1: Set Config Key
 
-In `./precompile/hello_world.go`, please search (`CTRL F`) `CUSTOM CODE STARTS HERE` to find the
-first area where we can modify the precompile.
+Let's jump to `helloworld/module.go` file first. This file contains the module definition for our
+precompile. We can see the `ConfigKey` is set to some default value of `helloWorldConfig`.
+This key should be unique to the precompile.
+This config key determines which JSON key to use when reading the precompile's config from the
+JSON upgrade/genesis file. In this case, the config key is `helloWorldConfig` and JSON config
+should look like:
 
-We can see our precompile address is set to some default value. We can cut the address from the var
-declaration block and remove it from the precompile.
+```json
+{
+  "helloWorldConfig": {
+    "blockTimestamp": 0
+		...
+  }
+}
+```
 
-![Singleton StatefulPrecompiledContract and signatures](./../../static/img/2022-09-01-22-46-00.png)
+### Step 2: Set Contract Address
 
-We can paste it in
-[`./precompile/params.go`](https://github.com/ava-labs/subnet-evm/blob/HelloWorldOfficialTutorial/precompile/params.go)
-and modify the default value to be the next user available stateful precompile address. For forks of
+We can see the `ContractAddress` is set to some default value.
+You should change that default value to a suitable address for your precompile.
```

Comment:
Should we be consistent between `we` and `you` and what person we are using?

Reply:
changed some places